cloud/google/clients/clients_test.go

```diff
@@ -0,0 +1,107 @@
+package clients_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"google.golang.org/api/googleapi"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sigs.k8s.io/cluster-api/cloud/google/clients"
+	"testing"
+)
+
+func createMuxAndServer() (*http.ServeMux, *httptest.Server) {
```

Comment:
note for the reviewer: with the exception of the "paginatedHandler(...)" function most of this code was copied from computeservice_test.go. It makes more sense to have it here if all of the various service_test.go files in this package will use it